### PR TITLE
DB-10201 Upgrade script for multi database

### PR DESCRIPTION
**Pull request by arnaud-splice**
### Description
This PR introduces an upgrade script to:

SYS.SYSDATABASES system table

Add a row to sys.sysdatabases with the first (original) database

Add column DATABASE_ID to SYS.SYSSCHEMAS

Add column DATABASE_ID to SYS.SYSROLES

Add column DATABASE_ID to SYS.SYSUSERS

Add 2 new views: SYSVW.SYSSEQUENCESVIEW, SYSVW.SYSDATABASESVIEW

Refresh existing views that are based on some of the system tables that were updated

### Testing
The update was tested manually. Further testing is part of DB-11130

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by ascend1 — APPROVED** [review]

**Comment by jyuanca** [comment]
Please hold on this PR. I'm rebasing and do more testing...

**Comment by arnaud-splice** [comment]
@jyuanca this PR would not go in master but in DB-10193
DB-10193 will most not likely not hit master for at least another 2 weeks

**Comment by arnaud-splice** [comment]
### Description
This PR introduces an upgrade script to: 

SYS.SYSDATABASES system table

Add a row to sys.sysdatabases with the first (original) database

Add column DATABASE_ID to SYS.SYSSCHEMAS

Add column DATABASE_ID to SYS.SYSROLES

Add column DATABASE_ID to SYS.SYSUSERS

Add 2 new views: SYSVW.SYSSEQUENCESVIEW, SYSVW.SYSDATABASESVIEW

Refresh existing views that are based on some of the system tables that were updated

### Testing
The update was tested manually. Further testing is part of DB-11130

**Comment by martinrupp** [comment]
hi, unfortunately, this fails upgrade, e.g.:

```
git checkout tags/3.2.0.2000
./start-splice-cluster
./sqlshell.sh
```
SQL
```
create table a ( i int);
create table b ( i int);
create trigger tg2 after insert on a  REFERENCING NEW AS NEW_ROW for each row when (NEW_ROW.i = 1) begin atomic insert into b values 1; insert into b values 2; end;
create trigger tg3 after insert on b  REFERENCING NEW AS NEW_ROW for each row when (NEW_ROW.i = 3) begin atomic insert into a values 1; insert into b values 2; end;
```
checkout this change and restart
```
./start-splice-cluster -k
git checkout origin/DB-10201
./start-splice-cluster -l
```

last command will fail to start the db again.

**Comment by martinrupp** [comment]
root cause looks to be in UpgradeScriptForTableColumnViewInSYSIBM accessing something that it shouldn't. Maybe UpgradeScriptToAddMultiDatabaseSupport needs to run before?

```
Caused by: java.lang.NullPointerException
	at java.io.StringReader.<init>(StringReader.java:50)
	at com.splicemachine.db.impl.services.uuid.BasicUUID.<init>(BasicUUID.java:78)
	at com.splicemachine.db.impl.services.uuid.BasicUUIDFactory.recreateUUID(BasicUUIDFactory.java:151)
	at com.splicemachine.db.impl.sql.catalog.SYSSCHEMASRowFactory.buildDescriptor(SYSSCHEMASRowFactory.java:234)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.getDescriptorViaIndexMinion(DataDictionaryImpl.java:8168)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.getDescriptorViaIndex(DataDictionaryImpl.java:8054)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.locateSchemaRowBody(DataDictionaryImpl.java:1162)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.locateSchemaRow(DataDictionaryImpl.java:1144)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.getSchemaDescriptorBody(DataDictionaryImpl.java:1322)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.getSchemaDescriptor(DataDictionaryImpl.java:1258)
	at com.splicemachine.db.iapi.sql.dictionary.SPSDescriptor.<init>(SPSDescriptor.java:190)
	at com.splicemachine.db.impl.sql.catalog.SYSSTATEMENTSRowFactory.buildDescriptor(SYSSTATEMENTSRowFactory.java:329)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.getDescriptorViaHeap(DataDictionaryImpl.java:8275)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.getAllSPSDescriptors(DataDictionaryImpl.java:4187)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.dropJDBCMetadataSPSes(DataDictionaryImpl.java:10851)
	at com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl.updateMetadataSPSes(DataDictionaryImpl.java:10876)
	at com.splicemachine.derby.impl.sql.catalog.upgrade.UpgradeScriptForTableColumnViewInSYSIBM.upgradeSystemTables(UpgradeScriptForTableColumnViewInSYSIBM.java:34)
```

full log [splice.log](https://github.com/splicemachine/spliceengine/files/6044150/splice.log)

File: splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceCatalogUpgradeScriptsTest.java [hunk 1]
```diff
@@ -50,7 +51,7 @@ private String replaceVersions(String s) {
     @Test
     public void test_since_1933()
     {
-        SpliceCatalogUpgradeScripts s = new SpliceCatalogUpgradeScripts(null, null);
+        SpliceCatalogUpgradeScripts s = new SpliceCatalogUpgradeScripts(null, null, new Properties());
```
[the remark this answers is not in the document]
Comment (arnaud-splice) [reply splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceCatalogUpgradeScriptsTest.java:54]: I do, this is added in the next PR that fixes ITs and unit tests.
See DB-11130

[the remark this answers is not in the document]
Comment (martinrupp) [reply splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceCatalogUpgradeScriptsTest.java:54]: ok

**Review by martinrupp — APPROVED** [review]

**Pull request merged**